src/main/java/org/elasticsearch/action/fieldstats/FieldStatsShardRequest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.fieldstats;
+
+import org.elasticsearch.action.support.broadcast.BroadcastShardOperationRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.io.IOException;
+
+/**
+ */
+public class FieldStatsShardRequest extends BroadcastShardOperationRequest {
+
+    private String[] fields;
+
+    public FieldStatsShardRequest() {
+    }
+
+    public FieldStatsShardRequest(ShardId shardId, FieldStatsRequest request) {
+        super(shardId, request);
+        this.fields = request.fields();
+    }
+
+    public String[] fields() {
```

Comment:
I hate to be that guy...but this is user facing stuff through java API. I think we should use the same convention in all java api classes. If we want to use regular getters, which I would be very happy with, I think we should go and make the change on all existing classes consistently. Otherwise, doesn't it become confusing for users to have some classes one way and some other classes another way?